opencode tool default model is not grok-code-fast-1

Full logs: https://gist.github.com/konard/815f4530aebdfc52a5eb652465183bf5

```
📝 Raw command:              
[2025-10-04T15:00:55.285Z] [INFO] (cd "/tmp/gh-issue-solver-1759590049343" && cat "/tmp/gh-issue-solver-1759590049343/opencode_prompt.txt" | opencode run --model anthropic/claude-3-5-sonnet)
[2025-10-04T15:00:55.286Z] [INFO] 
[2025-10-04T15:00:55.288Z] [INFO] 📋 Command details:          
[2025-10-04T15:00:55.289Z] [INFO]   📂 Working directory:      /tmp/gh-issue-solver-1759590049343
[2025-10-04T15:00:55.290Z] [INFO]   🌿 Branch:                 issue-398-5fa2baef
[2025-10-04T15:00:55.291Z] [INFO]   🤖 Model:                  OpenCode SONNET
[2025-10-04T15:00:55.292Z] [INFO] 
▶️ Streaming output:         

[2025-10-04T15:00:56.857Z] [INFO] [91m[1mError: [0mUnexpected error, check log file at /home/hive/.local/share/opencode/log/2025-10-04T150056.log for more details
[2025-10-04T15:00:56.860Z] [INFO] 

[2025-10-04T15:00:56.931Z] [ERROR] 

❌ Command failed: No output processed and errors detected in stderr
[2025-10-04T15:00:56.933Z] [ERROR]    [91m[1mError: [0mUnexpected error, check log file at /home/hive/.local/share/opencode/log/2025-10-04T150056.log for more details
[2025-10-04T15:00:56.961Z] [INFO] 
📈 System resources after execution:
[2025-10-04T15:00:56.962Z] [INFO]    Memory: MemFree:          215596 kB
[2025-10-04T15:00:56.963Z] [INFO]    Load: 0.64 0.16 0.05 1/163 75298
[2025-10-04T15:00:56.966Z] [INFO] 
[2025-10-04T15:00:56.967Z] [ERROR] ❌ OPENCODE execution failed
[2025-10-04T15:00:56.968Z] [INFO] 📁 Full log file: /home/hive/solve-2025-10-04T15-00-42-187Z.log
```

The command will not work without Grok Code Fast model, it is now the only free model we have access to for opencode tool.

--- a/experiments/test-model-default-fix.mjs
+++ b/experiments/test-model-default-fix.mjs
@@ -1,0 +1,56 @@
+#!/usr/bin/env node
+
+// Test script to verify the model default fix for opencode tool
+
+console.log('Testing model default fix for opencode tool...\n');
+
+// Test 1: Check if --model is in rawArgs
+const testCases = [
+  {
+    rawArgs: ['https://github.com/owner/repo/issues/123', '--tool', 'opencode'],
+    expected: { shouldSetDefault: true, tool: 'opencode' },
+    description: 'opencode tool without --model flag'
+  },
+  {
+    rawArgs: ['https://github.com/owner/repo/issues/123', '--tool', 'opencode', '--model', 'gpt4o'],
+    expected: { shouldSetDefault: false, tool: 'opencode' },
+    description: 'opencode tool with --model flag'
+  },
+  {
+    rawArgs: ['https://github.com/owner/repo/issues/123', '--tool', 'opencode', '-m', 'gpt4o'],
+    expected: { shouldSetDefault: false, tool: 'opencode' },
+    description: 'opencode tool with -m flag'
+  },
+  {
+    rawArgs: ['https://github.com/owner/repo/issues/123', '--tool', 'claude'],
+    expected: { shouldSetDefault: false, tool: 'claude' },
+    description: 'claude tool without --model flag'
+  },
+  {
+    rawArgs: ['https://github.com/owner/repo/issues/123', '--tool', 'opencode', '--verbose'],
+    expected: { shouldSetDefault: true, tool: 'opencode' },
+    description: 'opencode tool with other flags but no --model'
+  }
+];
+
+for (const testCase of testCases) {
+  console.log(`Test: ${testCase.description}`);
+  console.log(`  Raw args: ${testCase.rawArgs.join(' ')}`);
+
+  const rawArgsString = testCase.rawArgs.join(' ');
+  const modelExplicitlyProvided = rawArgsString.includes('--model') || rawArgsString.includes('-m');
+  const tool = testCase.rawArgs[testCase.rawArgs.indexOf('--tool') + 1];
+  const shouldSetDefault = tool === 'opencode' && !modelExplicitlyProvided;
+
+  console.log(`  Model explicitly provided: ${modelExplicitlyProvided}`);
+  console.log(`  Should set default: ${shouldSetDefault}`);
+  console.log(`  Expected: ${testCase.expected.shouldSetDefault}`);
+
+  if (shouldSetDefault === testCase.expected.shouldSetDefault) {
+    console.log('  ✅ PASS\n');
+  } else {
+    console.log('  ❌ FAIL\n');
+  }
+}
+
+console.log('All tests completed!');

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@deep-assistant/hive-mind",
-  "version": "0.21.8",
+  "version": "0.21.9",
   "description": "AI-powered issue solver and hive mind for collaborative problem solving",
   "main": "src/hive.mjs",
   "type": "module",

--- a/src/solve.config.lib.mjs
+++ b/src/solve.config.lib.mjs
@@ -169,5 +169,18 @@ export const createYargsConfig = (yargsInstance) => {
 // Parse command line arguments - now needs yargs and hideBin passed in
 export const parseArguments = async (yargs, hideBin) => {
   const rawArgs = hideBin(process.argv);
-  return await createYargsConfig(yargs(rawArgs)).argv;
+  const argv = await createYargsConfig(yargs(rawArgs)).argv;
+
+  // Post-processing: Fix model default for opencode tool
+  // Yargs doesn't properly handle dynamic defaults based on other arguments,
+  // so we need to handle this manually after parsing
+  const rawArgsString = rawArgs.join(' ');
+  const modelExplicitlyProvided = rawArgsString.includes('--model') || rawArgsString.includes('-m');
+
+  if (argv.tool === 'opencode' && !modelExplicitlyProvided) {
+    // User did not explicitly provide --model, so use the correct default for opencode
+    argv.model = 'grok-code-fast-1';
+  }
+
+  return argv;
 };

